[Design2-118, Design2-119, Design2-120] Copied common folder, font.scss, closeButton, FontAwesomeV6Icon, Typography to DSCO package

levadadenys opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->

### [Design2-118, Design2-119, Design2-120] Copied common folder, font.scss, closeButton, FontAwesomeV6Icon, Typography to DSCO package

- feat(dsco/common): copied common folder to dsco package
- feat(changelog-pre1): created CHANGELOG-pre1.md
- feat(dsco/common/styles): coppied font.scss to dsco/common/styles
- feat(dsco/typography): coppied typography to dsco/typography
- feat(dsco/fontAwesomeV6Icon): copied FontAwesomeV6Icon to dsco/fontAwesomeV6Icon
- feat(dsco/closeButton): copied CloseButton to dsco/closeButton
- fix: fixed import path support fot test files


https://github.com/user-attachments/assets/a38021ef-9685-4f30-bfb0-725d74df0731



## Links

- jira ticket: [Design2-118](https://codedotorg.atlassian.net/browse/DESIGN2-218), [Design2-119](https://codedotorg.atlassian.net/browse/DESIGN2-219), [Design2-120](https://codedotorg.atlassian.net/browse/DESIGN2-220)

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

<!--
- spec: []()

-->

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

bethanyaconnor commented on frontend/packages/dsco/CHANGELOG-pre1.md:
```diff
@@ -0,0 +1,167 @@
+
+## `componentLibrary/typography`
+
+## [2.0.2](https://github.com/code-dot-org/code-dot-org/pull/62837)
```
Not blocking but this ordering is confusing me. The versions are descending then jump up to 2.0.2 and descend again. Should these be moved to the top of this file so all the versions are strictly descending?

levadadenys replied:
Good question!
This Changelog-pre1.md will be a place where all previous Changelogs will be combined into 1, sorted per component. (Because initially we had a separate Changelog for every component).
pre-1 versions of the package might also end up in this file (on the top of the file).
Starting from v1.0.0.alpha0 version all the changelogs will be in a separate, main CHANGELOG.md so that those versions will be sorted in regular pattern everyone uses everywhere.

Also worth mentioning here that Typography is the only component that has v1 and v2 (major versions) since it was the first component we made and since it was used slightly different

bethanyaconnor commented:
All of this is just copied from what's in apps/src/componentLibrary currently right? Or were there other changes made?

levadadenys commented:
> All of this is just copied from what's in apps/src/componentLibrary currently right? Or were there other changes made?

Coppied, fixed lint warnings, updated structure to match what we decided to use for dsco package, Changelogs moved to CHANGELOG-pre1.md, updated import statements in some places. The logic and styles of the components are intended unchanged comparing to apps/src/componentLibrary implementation

bencodeorg commented:
General question now that we're actually starting to copy components -- is the plan to pause any new development on any component that gets copied over here until `apps` is consuming this code? Or to apply changes in both places?

bencodeorg commented:
🎉 🎉 🎉 

![image](https://github.com/user-attachments/assets/dc9f2961-afe2-4e3e-a8c5-be58bc45c89c)

bencodeorg commented on frontend/packages/dsco/src/typography/README.md:
```diff
@@ -0,0 +1,94 @@
+
+This is a set of Typography components for use in the Code.org component library.
+Implements Typography components with our rebranded styles. Part of global rebranding.
+At some point it will be moved to (https://github.com/code-dot-org/dsco_), however the signature of the components should
```
I like keeping this as a straight copy of what's currently in `apps`, but just making a note we should update this later.

bencodeorg commented on frontend/packages/dsco/src/fontAwesomeV6Icon/stories/FontAwesomeV6Icon.story.tsx:
```diff
@@ -0,0 +1,169 @@
+import FontAwesomeV6Icon, {FontAwesomeV6IconProps} from '../index';
+
+export default {
+  title: 'DesignSystem/FontAwesomeV6Icon',
```
Also not a thing to fix right now, but not sure if we need the "Design System" category in Storybook any more.

levadadenys replied:
Correct. I keep it same way is to was before just to save some time, plus we'll probably need to replace it with atom/molecule/organism soon, so since it'll be updated anyways decided to keep as is for now

bencodeorg reviewed (approved):
LGTM!

To Bethany's question about "what's in here?", I think it would be easier in the future to review if you broke this into a few separate PRs (I think maybe one per component, and one for the common folder stuff you copied)? I ended up going though commit by commit, which worked well enough.

levadadenys commented:
> General question now that we're actually starting to copy components -- is the plan to pause any new development on any component that gets copied over here until apps is consuming this code? Or to apply changes in both places?

Good question! Ideally we shouldn't make any updates untill all the components get moved to dsco package and consumed by code-dot-org. But in case there will be some urgent updates - we should probably make these updates in both instances
